File: app/src/main/java/com/andraganoid/playsomemedia/model/AudioDao.java
package com.andraganoid.playsomemedia.model;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import java.util.List;

@Dao
public interface AudioDao {

    @Query("SELECT * FROM audio_table ORDER BY artist AND title ASC")
    LiveData <List <Audio>> getAllAudios();

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    void insert(Audio audio);

    @Query("DELETE FROM audio_table")
    void deleteAllAudio();

    @Delete
    void deleteAudio(Audio audio);
}

File: app/src/main/java/com/andraganoid/playsomemedia/k/db/AudioDao.kt
package com.andraganoid.playsomemedia.k.db

import androidx.lifecycle.LiveData
import androidx.room.*
import com.andraganoid.playsomemedia.model.Audio

@Dao
interface AudioDao {
    @Query("SELECT * FROM AUDIO_TABLE ORDER BY artist AND title ASC")
    fun getAllAudios(): LiveData<List<Audio?>?>?

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    fun insert(audio: Audio?)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAudioList(audioList:List<Audio?>?)

    @Query("DELETE FROM AUDIO_TABLE")
    fun deleteAllAudios()

    @Delete
    fun deleteAudio(audio: Audio?)
}